### PR TITLE
Maint/master/initial commits

### PR DESCRIPTION
**Pull request by melissa**

### CODE REVIEW

File: .fixtures.yml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+# This file can be used to install module depdencies for unit testing
+# See https://github.com/puppetlabs/puppetlabs_spec_helper#using-fixtures for details
+---
+fixtures:
+  forge_modules:
+#     stdlib: "puppetlabs/stdlib"
```
Comment (melissa) [inline .fixtures.yml]: This code has a dependency on `puppet/provider/parsedfile`, but it doesn't seem like we can express this dependency here

File: spec/spec_helper_acceptance.rb [hunk 2]
```diff
@@ -0,0 +1,24 @@
+require 'puppet'
+require 'beaker-rspec'
+require 'beaker/module_install_helper'
+require 'beaker/puppet_install_helper'
+
+$LOAD_PATH << File.join(__dir__, 'acceptance/lib')
+
+def beaker_opts
+  { debug: true, trace: true, expect_failures: true, acceptable_exit_codes: (0...256) }
+end
```
Comment (joshcooper) [inline spec/spec_helper_acceptance.rb]: I have the same in the zone module, but it doesn't seem to do anything...

File: .project [hunk 3]
```diff
@@ -0,0 +1,23 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<projectDescription>
+	<name>puppetlabs-mailalias</name>
```
Comment (joshcooper) [inline .project]: puppetlabs-mailalias_core

File: metadata.json [hunk 4]
```diff
@@ -0,0 +1,92 @@
+{
+  "name": "puppetlabs-mailalias",
+  "version": "0.1.0",
+  "author": "Puppet Labs",
+  "summary": "Creates an email alias in the local alias database.",
+  "license": "Apache-2.0",
+  "source": "https://github.com/puppetlabs/puppetlabs-mailalias",
+  "project_page": "https://github.com/puppetlabs/puppetlabs-mailalias/blob/master/README.md",
```
Comment (joshcooper) [inline metadata.json]: same here

File: spec/lib/puppet_spec/files.rb [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+  def expect_file_mode(file, mode)
+    actual_mode = '%o' % Puppet::FileSystem.stat(file).mode
+    target_mode = if Puppet.features.microsoft_windows?
+                    mode
+                  else
+                    '10' + '%04i' % mode.to_i
+                  end
+    expect(actual_mode).to eq(target_mode)
+  end
+end
```
Comment (joshcooper) [inline spec/lib/puppet_spec/files.rb]: I'm wondering if rspec-puppet already has helpers for managing temp files and directories?

Comment (joshcooper) [reply spec/lib/puppet_spec/files.rb]: I didn't see any, nvm

**Comment by kris-bosland** [comment]
Should we check in the PDK skeleton first, to reduce the boilerplate in the diff of the initial commit?

**Comment by melissa** [comment]
@kris-bosland if you review this commit by commit, it will be a lot easier to see the changes I've made.

**Comment by joshcooper** [comment]
Minor issue, I'm getting rubocop errors when running the rubocop task directly. I'm having pdk issues at the moment, so can't run the `pdk validate` command.

```
$ bundle exec rake rubocop
Running RuboCop...
Inspecting 12 files
.C.CCC....C.

Offenses:

lib/puppet/type/mailalias.rb:5:1: C: Layout/IndentHeredoc: Use 2 spaces for indentation in a heredoc by using some library(e.g. ActiveSupport's String#strip_heredoc). (https://github.com/bbatsov/ruby-style-guide#squiggly-heredocs)
Creates an email alias in the local alias database. ...
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
spec/acceptance/tests/destroy_spec.rb:15:36: C: Layout/ExtraSpacing: Unnecessary spacing detected.
      on(agent, 'cat /etc/aliases')  do |res|
                                   ^
spec/acceptance/tests/modify_spec.rb:16:36: C: Layout/ExtraSpacing: Unnecessary spacing detected.
      on(agent, 'cat /etc/aliases')  do |res|
                                   ^
spec/acceptance/tests/modify_spec.rb:37:36: C: Layout/ExtraSpacing: Unnecessary spacing detected.
      on(agent, 'cat /etc/aliases')  do |res|
                                   ^
spec/acceptance/tests/query_spec.rb:15:36: C: Layout/ExtraSpacing: Unnecessary spacing detected.
      on(agent, 'cat /etc/aliases')  do |res|
                                   ^
spec/spec_helper_acceptance.rb:13:3: C: Style/InverseMethods: Use reject instead of inverting select.
  agents.select { |agent| !agent['platform'].include?('windows') }
  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

12 files inspected, 6 offenses detected
RuboCop failed!
```

**Comment by melissa** [comment]
@joshcooper updated

File: metadata.json [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+      ]
+    },
+    {
+      "operatingsystem": "Ubuntu",
+      "operatingsystemrelease": [
+        "16.04"
+      ]
+    },
+    {
+      "operatingsystem": "windows",
```
Comment (melissa) [inline metadata.json]: I probably need to refine this list of platforms, since I'm not sure sendmail is available on windows...

Comment (joshcooper) [reply metadata.json]: 👍 to dropping windows

**Comment by melissa** [comment]
screen shot 2018-04-26 at 11 08 44 am


I ran the adhoc pipeline set up with (https://github.com/puppetlabs/ci-job-configs/pull/4369), and it looks good!

**Pull request merged**